examples/preprocessing/plot_all_scaling.py

```diff
@@ -0,0 +1,328 @@
+            ax_colorbar)
+
+
+def plot_distribution(axes, X, y, hist_nbins=50, title="",
+                      x0_label="", x1_label=""):
+    ax, hist_X1, hist_X0 = axes
+
+    ax.set_title(title)
+    ax.set_xlabel(x0_label)
+    ax.set_ylabel(x1_label)
+
+    # The scatter plot
+    colors = cm.plasma_r(y)
+    ax.scatter(X[:, 0], X[:, 1], alpha=0.5, marker='o', s=5, lw=0, c=colors)
+
+    # Removing the top and the right spine for aesthetics
+    # make nice axis layout
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.get_xaxis().tick_bottom()
+    ax.get_yaxis().tick_left()
+    ax.spines['left'].set_position(('outward', 10))
+    ax.spines['bottom'].set_position(('outward', 10))
+
+    # Histogram for axis X1 (feature 5)
+    hist_X1.set_ylim(ax.get_ylim())
+    hist_X1.hist(X[:, 1], bins=hist_nbins, orientation='horizontal',
+                 color='grey', ec='grey')
+    hist_X1.axis('off')
+
+    # Histogram for axis X0 (feature 0)
+    hist_X0.set_xlim(ax.get_xlim())
+    hist_X0.hist(X[:, 0], bins=hist_nbins, orientation='vertical',
+                 color='grey', ec='grey')
+    hist_X0.axis('off')
+
+###############################################################################
+# Two plots will be shown for each scaler/normalizer/transformer. The left
+# figure will show a scatter plot of the full data set while the right figure
+# will exclude the extreme values considering only 99 % of the data set,
+# excluding marginal outliers. In addition, the marginal distributions for each
+# feature will be shown on the side of the scatter plot.
+
+
+def make_plot(item_idx):
+    _, X = list(distributions.items())[item_idx]
+    ax_zoom_out, ax_zoom_in, ax_colorbar = create_axes()
+    axarr = (ax_zoom_out, ax_zoom_in)
+    plot_distribution(axarr[0], X, y, hist_nbins=200,
+                      x0_label="Median Income",
+                      x1_label="Number of households")
+
+    # zoom-in
+    zoom_in_percentile_range = (0, 99)
+    cutoffs_X0 = np.percentile(X[:, 0], zoom_in_percentile_range)
+    cutoffs_X1 = np.percentile(X[:, 1], zoom_in_percentile_range)
+
+    non_outliers_mask = (
+        np.all(X > [cutoffs_X0[0], cutoffs_X1[0]], axis=1) &
+        np.all(X < [cutoffs_X0[1], cutoffs_X1[1]], axis=1))
+    plot_distribution(axarr[1], X[non_outliers_mask], y[non_outliers_mask],
+                      hist_nbins=50,
+                      x0_label="Median Income",
+                      x1_label="Number of households")
+
+    norm = mpl.colors.Normalize(y_full.min(), y_full.max())
+    mpl.colorbar.ColorbarBase(ax_colorbar, cmap=cm.plasma_r,
+                              norm=norm, orientation='vertical',
+                              label='Color mapping for values of y')
+
+
+########################################################################
+# .. _results:
+#
+# Original data
+# -------------
+#
+# The following plot displays the original data distribution in the left panel
```

Comment:
I think you need to start by saying that each transformation is plotted for two features, with the left plot showing the entire dataset, and the right zoomed in to show ....